config/migrate.ts

```diff
@@ -0,0 +1,209 @@
+  const updatedConfig = {
+    ...rest,
+    defaultAccount: defaultPortal,
+    accounts: portals
+      .filter(({ portalId }) => portalId !== undefined)
+      .map(({ portalId, ...rest }) => ({
+        ...rest,
+        accountId: portalId!,
+      })),
+  };
+  writeGlobalConfigFile(updatedConfig);
+}
+
+type ConflictValue = boolean | string | number | CmsPublishMode | Environment;
+export type ConflictProperty = {
+  property: keyof CLIConfig_NEW;
+  oldValue: ConflictValue;
+  newValue: ConflictValue;
+};
+
+export function mergeConfigProperties(
+  globalConfig: CLIConfig_NEW,
+  deprecatedConfig: CLIConfig_DEPRECATED,
+  force?: boolean
+): {
+  initialConfig: CLIConfig_NEW;
+  conflicts: Array<ConflictProperty>;
+} {
+  const propertiesToCheck: Array<keyof Partial<CLIConfig>> = [
+    DEFAULT_CMS_PUBLISH_MODE,
+    HTTP_TIMEOUT,
+    ENV,
+    HTTP_USE_LOCALHOST,
+  ];
+  const conflicts: Array<ConflictProperty> = [];
+
+  propertiesToCheck.forEach(prop => {
+    if (prop in globalConfig && prop in deprecatedConfig) {
+      if (force) {
+        // @ts-expect-error Cannot reconcile CLIConfig_NEW and CLIConfig_DEPRECATED types
+        globalConfig[prop] = deprecatedConfig[prop];
+      } else if (globalConfig[prop] !== deprecatedConfig[prop]) {
+        conflicts.push({
+          property: prop,
+          oldValue: deprecatedConfig[prop]!,
+          newValue: globalConfig[prop]!,
+        });
+      }
+    } else {
+      // @ts-expect-error Cannot reconcile CLIConfig_NEW and CLIConfig_DEPRECATED types
+      globalConfig[prop] = deprecatedConfig[prop];
+    }
+  });
+
+  if (
+    DEFAULT_ACCOUNT in globalConfig &&
+    DEFAULT_PORTAL in deprecatedConfig &&
+    globalConfig.defaultAccount !== deprecatedConfig.defaultPortal
+  ) {
+    conflicts.push({
+      property: DEFAULT_ACCOUNT,
+      oldValue: deprecatedConfig.defaultPortal!,
+      newValue: globalConfig.defaultAccount!,
+    });
+  } else if (DEFAULT_PORTAL in deprecatedConfig) {
+    globalConfig.defaultAccount = deprecatedConfig.defaultPortal;
+  }
+
+  return { initialConfig: globalConfig, conflicts };
+}
+
+function mergeAccounts(
+  globalConfig: CLIConfig_NEW,
+  deprecatedConfig: CLIConfig_DEPRECATED
+): {
+  finalConfig: CLIConfig_NEW;
+  skippedAccountIds: Array<string | number>;
+} {
+  let existingAccountIds: Array<string | number> = [];
+  if (globalConfig.accounts && deprecatedConfig.portals) {
+    existingAccountIds = globalConfig.accounts.map(
+      account => account.accountId
+    );
+
+    const newAccounts = deprecatedConfig.portals
+      .filter(portal => !existingAccountIds.includes(portal.portalId!))
+      .map(({ portalId, ...rest }) => ({
+        ...rest,
+        accountId: portalId!,
+      }));
+
+    if (newAccounts.length > 0) {
+      globalConfig.accounts.push(...newAccounts);
+    }
+  }
+
+  return {
+    finalConfig: globalConfig,
+    skippedAccountIds: deprecatedConfig.portals
```

Comment:
Not a blocker, but you could DRY this up a little by generating the `skippedAccounIds` array in the `filter()`